src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,136 @@
+package duke.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.Command;
+import duke.commands.CommandType;
+import duke.tasks.DeadlineTask;
+import duke.tasks.EventTask;
+import duke.tasks.ToDoTask;
+import duke.ui.Ui;
+
+/**
+ * The Parser class is responsible for parsing user input commands and converting them into executable commands.
+ * It supports commands such as adding tasks, marking tasks as done, deleting tasks, and more.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and returns an appropriate Command object based on the input.
+     *
+     * @param userInput The user's input command as a string.
+     * @return A command object that corresponds to the parsed input.
+     */
+    public static Command parse(String userInput) {
```

Comment:
Might it be beneficial to break down the parsing of different commands into their own methods for readability and maintainability?